web-js/src/components/WelcomePage.js

```diff
@@ -0,0 +1,146 @@
+  const clearErrors = () => setError(noErrors);
+  // Manage password visibility
+  const [showPassword, setShowPassword] = React.useState(false);
+  const toggleShowPassword = () => setShowPassword(toggleBoolean);
+
+  const onFormSubmit = async ({ email, password }) => {
+    clearErrors();
+    try {
+      if (isSignup) {
+        await realmApp.emailPasswordAuth.registerUser(email, password);
+      }
+      await realmApp.logIn(Realm.Credentials.emailPassword(email, password));
+    } catch (err) {
+      handleAuthenticationError(err, setError);
+    }
+  };
+
+  return (
+    <Container maxWidth="sm" className="main-container">
+      <Card className="auth-card" variant="outlined">
+        <form
+          className="auth-form"
+          onSubmit={(e) => {
+            e.preventDefault();
+            function getFormValues(htmlFormElement) {
+              const formData = new FormData(htmlFormElement);
+              return Object.fromEntries(formData.entries());
+            }
+            const { email, password } = getFormValues(e.target);
+            onFormSubmit({ email, password });
+          }}
+        >
+          <Typography component="h2" variant="h4" gutterBottom>
+            {isSignup ? "Sign Up" : "Log In"}
+          </Typography>
+          <TextField
+            id="input-email"
+            name="email"
+            label="Email Address"
+            variant="outlined"
+            error={Boolean(error.email)}
+            helperText={error.email ?? ""}
+          />
+          <TextField
+            id="input-password"
+            type={showPassword ? "text" : "password"}
+            name="password"
+            label="Password"
+            variant="outlined"
+            error={Boolean(error.password)}
+            helperText={error.password ?? ""}
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position="end">
+                  <IconButton
+                    aria-label="toggle password visibility"
+                    onClick={toggleShowPassword}
+                    onMouseDown={(e) => {
+                      e.preventDefault();
+                    }}
+                  >
+                    {showPassword ? <Visibility /> : <VisibilityOff />}
+                  </IconButton>
+                </InputAdornment>
+              )
+            }}
+          />
+          <Button type="submit" variant="contained" color="primary">
+            {isSignup ? "Create Account" : "Log In"}
+          </Button>
+          <button
+            type="button"
+            className="link-button"
+            onClick={() => toggleIsSignup()}
+          >
+            {isSignup
+              ? "Already have an account? Log In"
+              : "Sign up for an account"}
+          </button>
+        </form>
+      </Card>
+      <MoreInfoTemplateAndDocs />
+    </Container>
+  );
+}
+
+function handleAuthenticationError(err, setError) {
```

Comment:
What happens if the `err` is not a `MongoDBRealmError` (like a network level error or something)? I believe the error would be swallowed, since the `err.error` and `err.statusCode` will both be `undefined` and the `default` case doesn't call `setError`.